Source/loadsave.cpp

```diff
@@ -1285,8 +1297,10 @@ static void SaveItem(SaveHelper *file, ItemStruct *pItem)
 	}
 
 	file->WriteLE<int32_t>(pItem->_iSeed);
-	file->WriteLE<int16_t>(pItem->_iCreateInfo);
-	file->Skip(2); // Alignment
+
+	file->WriteLE<uint16_t>(pItem->_iCreateInfo);
+	file->Skip(2); // Alignment, _iCreateInfo was written as a 4 byte LE value but only requires 2 bytes

```

Comment:
I don't think this is why alignment happen for structs https://en.m.wikipedia.org/wiki/Data_structure_alignment

Maybe you are not aware but the original save files would mostly just memcpy structs to the file buffer

Reply:
This was a bit of a guess based on the comments in devilution for certain structs:
https://github.com/diasurgical/devilution/blob/e426c38adb5df05b8e2c02a607529bdc913c281b/structs.h#L241-L247

Situations like this looked more like type conversion/aliasing? compared to other instances where a group of short/char variables needed padding to get back to 4 byte alignment.

Reply:
I think that comment is wrong, the compiler never was able to do this when we worked with it and this looks more like just regular padding.

Reply:
Ahh, in that case I'll revert those comment changes. The other comments I updated with `packed _var, _var, ...` were also more helpers for myself along the same lines so I'll revert those as well.